pkg/gvisor/enable.go

```diff
@@ -0,0 +1,208 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gvisor
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"k8s.io/minikube/pkg/minikube/constants"
+
+	"github.com/pkg/errors"
+	"k8s.io/minikube/pkg/minikube/assets"
+)
+
+const (
+	nodeDir = "/node"
+)
+
+// Enable follows these steps for enabling gvisor in minikube:
+//   1. creates necessary directories for storing binaries and runsc logs
+//   2. downloads runsc and gvisor-containerd-shim
+//   3. copies necessary containerd config files
+//   4. restarts containerd
+func Enable() error {
+	if err := makeDirs(); err != nil {
+		return errors.Wrap(err, "creating directories on node")
+	}
+	if err := downloadBinaries(); err != nil {
+		return errors.Wrap(err, "downloading binaries")
+	}
+	if err := copyFiles(); err != nil {
+		return errors.Wrap(err, "copying files")
+	}
+	if err := restartContainerd(); err != nil {
+		return errors.Wrap(err, "restarting containerd")
+	}
+	// sleep for one year so the pod continuously runs
+	time.Sleep(24 * 7 * 52 * time.Hour)
+	return nil
+}
+
+// makeDirs creates necessary directories on the node
+func makeDirs() error {
+	// Make /run/containerd/runsc to hold logs
+	fp := filepath.Join(nodeDir, "run/containerd/runsc")
+	if err := os.MkdirAll(fp, 0755); err != nil {
+		return errors.Wrap(err, "creating runsc dir")
+	}
+
+	// Make /usr/local/bin to store the runsc binary
+	fp = filepath.Join(nodeDir, "usr/local/bin")
+	if err := os.MkdirAll(fp, 0755); err != nil {
+		return errors.Wrap(err, "creating usr/local/bin dir")
+	}
+
+	// Make /tmp/runsc to also hold logs
+	fp = filepath.Join(nodeDir, "tmp/runsc")
+	if err := os.MkdirAll(fp, 0755); err != nil {
+		return errors.Wrap(err, "creating runsc logs dir")
+	}
+
+	return nil
+}
+
+func downloadBinaries() error {
+	if err := runsc(); err != nil {
+		return errors.Wrap(err, "downloading runsc")
+	}
+	if err := gvisorContainerdShim(); err != nil {
+		return errors.Wrap(err, "downloading gvisor-containerd-shim")
+	}
+	return nil
+}
+
+// downloads the gvisor-containerd-shim
+func gvisorContainerdShim() error {
+	dest := filepath.Join(nodeDir, "usr/bin/gvisor-containerd-shim")
+	// TODO: priyawadhwa@, replace with official release
+	return downloadFileToDest("http://storage.googleapis.com/balintp-minikube/gvisor-containerd-shim", dest)
```

Comment:
It may be worth making these download URLs into consts

Reply:
done